grpc/src/main/java/io/stargate/grpc/service/interceptors/NewConnectionInterceptor.java

```diff
@@ -83,13 +83,14 @@ public NewConnectionInterceptor(
               .headers(stringHeaders)
               .remoteAddress(call.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR))
               .build();
+      Context context = Context.current().withValue(GrpcService.HEADERS_KEY, stringHeaders);
 
-      Connection connection = connectionCache.get(info);
+      if (!"stargate.Stargate/getSchemaChanges"
```

Comment:
why do we need this special condition?

Reply:
That operation interacts with the `Persistence` directly, it doesn't need the `Connnection`. So the idea was to avoid creating it.
But maybe that's overkill (see my other comment).